cms/djangoapps/contentstore/management/commands/sync_courses.py

```diff
@@ -0,0 +1,68 @@
+"""
+Sync courses from catalog service.  This is used to setup a master's
+integration environment.
+"""
+import logging
+from six import text_type
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+
+from contentstore.management.commands.utils import user_from_str
+from contentstore.views.course import create_new_course_in_store
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.catalog.utils import get_course_runs
+from xmodule.modulestore.exceptions import DuplicateCourseError
+from xmodule.modulestore import ModuleStoreEnum
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Command to populate modulestore with courses from the discovery service.
+
+    Example: ./manage.py cms sync_courses staff@example.com
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('user')
+
+    def get_user(self, user):
+        """
+        Return a User object.
+        """
+        try:
+            user_object = user_from_str(user)
+        except User.DoesNotExist:
+            raise CommandError(u"No user {user} found.".format(user=user))
+        return user_object
+
+    def handle(self, *args, **options):
+        """Execute the command"""
+        user = self.get_user(options['user'])
+
+        course_runs = get_course_runs()
```

Comment:
You should have an exception handle here for the case where catalog integration is not setup in LMS. There are plenty of openedX instance where course-discovery service is not setup.

Reply:
If there's no configuration for course-discovery the underlying function will just return an empty array.